File: greatest-common-divisor/greatest-common-divisor.cpp
// Euclid’s algorithm for finding the greatest common divisor of
// two numbers
#include <iostream>
using namespace std;

int gcd(int a, int b) {
    if (b == 0) return a;
    int r = a % b;
    return gcd(b, r);
}

int main() {
    int x, y;
    cout << "Enter a first number: ";
    cin >> x;
    cout << "Enter a second number: ";
    cin >> y;
    cout << "Answer is: " << gcd(x, y);
}

File: greatest-common-divisor/greatest_common_divisor.py
# Euclid’s algorithm for finding the greatest common divisor of
# two numbers
def gcd(a, b):
    if b == 0:
        return a
    r = a % b
    return gcd(b, r)


if __name__ == "__main__":
    x = int(input("Enter a first number: "))
    y = int(input("Enter a second number: "))
    print("Answer is: ", gcd(x, y))
